frontend/src/stores/FilterQueryStore.ts

```diff
@@ -1,17 +1,127 @@
 
 /**
- * Contains the current state of the filter.
+ * Contains the current state of the filterQuery.
  */
 @injectable()
 export default class FilterQueryStore extends SimpleStore<FilterQuery> {
+  @inject(FilterStateStore)
+  private readonly filterStateStore!: FilterStateStore;
+
   protected getInitialValue(): FilterQuery {
     return {
       limits: { edges: 150, nodes: 200 },
       includeSubsidiary: true,
     };
   }
+
+  /**
+   * Updates the state of the filterQuery by synchronizing it with the {@link FilterStateStore}.
+   */
+  update(): void {
+    const filter = this.filterStateStore.getValue();
+    const filterQuery = this.convertToFilterQuery(filter);
+    this.mergeState(filterQuery);
+  }
+
+  /**
+   * Converts a {@link FilterState} given from the users input to a {@link FilterQuery}.
+   * @param filterState - the {@link FilterState} to be converted
+   * @returns {@link FilterQuery} as conversion result
+   * @private
+   */
+  private convertToFilterQuery(filterState: FilterState): FilterQuery {
+    const createEntityCondition = (filterLineStates: FilterLineState[]) => {
+      const entityConditions: FilterCondition[] = [];
+
+      for (const filterLineState of filterLineStates) {
+        // collect from filter line
+
+        if (filterLineState.isActive) {
+          const propertyConditions: FilterCondition[] =
+            this.convertToPropertyConditions(filterLineState.propertyFilters);
+
+          // add the ofTypeCondition to the entity
+          const ofTypeCondition = OfTypeCondition(filterLineState.type);
+          const lineCondition: FilterCondition =
+            propertyConditions.length > 0
+              ? MatchAllCondition(
+                  ofTypeCondition,
+                  MatchAllCondition(...propertyConditions)
+                )
+              : ofTypeCondition;
+
+          entityConditions.push(lineCondition);
+        }
+      }
+      return MatchAnyCondition(...entityConditions);
+    };
+
+    return {
+      filters: {
+        nodes: createEntityCondition(filterState.nodes),
+        edges: createEntityCondition(filterState.edges),
+      },
+    };
+  }
+
+  /**
+   * Converts an Array of raw {@link FilterPropertyState}s given from the users input
+   * in a {@link FilterLineProperties} to {@link FilterCondition}s representing the properties of an entity.
+   * @param propertyFilters - the {@link FilterPropertyState}s to be converted
+   * @returns {@link FilterCondition}s representing the properties of an entity.
+   * @private
+   */
+  private convertToPropertyConditions(
+    propertyFilters: FilterPropertyState[]
+  ): FilterCondition[] {
+    const propertyConditions: FilterCondition[] = [];
+
+    for (const propertyFilter of propertyFilters) {
+      // collect from dialog
+      // There is a filter specified for the property
+
+      if (propertyFilter.values !== null && propertyFilter.values.length > 0) {
```

Comment:
[Optional improvement/Hint] Use [Optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).
```suggestion
      if (propertyFilter.values?.length > 0) {
```